packages/client/hmi-client/src/components/model/tera-model.vue

```diff
@@ -163,28 +183,28 @@ const optionsMenuPt = {
 	}
 };
 
-async function updateModelContent(updatedModel: Model) {
-	if (!useProjects().hasEditPermission()) {
-		logger.error('You do not have permission to edit this model.'); // FIXME: Disable asset editing options if user does not have permission
-		return;
-	}
-	await updateModel(updatedModel);
-	logger.info('Saved changes.');
+async function updateModelContent() {
+	if (!hasEditPermission.value || !temporaryModel.value) return;
+	await updateModel(temporaryModel.value);
+	logger.info('Changes to the model has been saved.');
 	await useProjects().refresh();
-	fetchModel();
+	await fetchModel();
 }
 
 async function updateModelName() {
-	if (model.value && !isEmpty(newName.value)) {
-		const modelClone = cloneDeep(model.value);
-		modelClone.header.name = newName.value;
-		await updateModelContent(modelClone);
+	if (temporaryModel.value && !isEmpty(newName.value)) {
+		temporaryModel.value.header.name = newName.value;
```

Comment:
Maybe check with @pascaleproulx  about this. I think she liked the "auto save" when renaming an asset during our meeting yesterday.

Reply:
Discussion moved to Slack: https://unchartedsoftware.slack.com/archives/C04L3FRD2BG/p1727279383143919